File: app/src/main/java/fr/corenting/edcompanion/fragments/SystemStationsFragment.java
package fr.corenting.edcompanion.fragments;

import org.greenrobot.eventbus.Subscribe;
import org.greenrobot.eventbus.ThreadMode;

import fr.corenting.edcompanion.activities.SystemDetailsActivity;
import fr.corenting.edcompanion.adapters.SystemStationsAdapter;
import fr.corenting.edcompanion.models.events.SystemStations;
import fr.corenting.edcompanion.utils.NotificationsUtils;

public class SystemStationsFragment extends AbstractListFragment<SystemStationsAdapter> {

    public static final String SYSTEM_STATIONS_FRAGMENT_TAG = "system_stations";

    public SystemStationsFragment() {
        loadDataOnCreate = false;
    }

    @Override
    void getData() {
        SystemDetailsActivity parent = ((SystemDetailsActivity) getActivity());
        parent.getData();
    }

    @Override
    SystemStationsAdapter getAdapter() {
        return new SystemStationsAdapter(getContext(), recyclerView);
    }

    @Subscribe(threadMode = ThreadMode.MAIN)
    public void onStationsEvent(SystemStations stations) {
        // Error case
        if (!stations.getSuccess()) {
            endLoading(true);
            NotificationsUtils.displayDownloadErrorSnackbar(getActivity());
            return;
        }

        endLoading(stations.getStations().size() == 0);
        recyclerViewAdapter.submitList(stations.getStations());
    }
}

File: app/src/main/java/fr/corenting/edcompanion/fragments/SystemStationsFragment.kt
package fr.corenting.edcompanion.fragments

import fr.corenting.edcompanion.activities.SystemDetailsActivity
import fr.corenting.edcompanion.adapters.SystemStationsAdapter
import fr.corenting.edcompanion.models.events.SystemStations
import fr.corenting.edcompanion.utils.NotificationsUtils
import org.greenrobot.eventbus.EventBus
import org.greenrobot.eventbus.Subscribe
import org.greenrobot.eventbus.ThreadMode

class SystemStationsFragment : AbstractListFragment<SystemStationsAdapter>() {
    init {
        loadDataOnCreate = false
    }

    override fun getNewRecyclerViewAdapter(): SystemStationsAdapter {
        return SystemStationsAdapter(context, binding.recyclerView)
    }


    override fun onStart() {
        super.onStart()

        EventBus.getDefault().register(this)
    }

    override fun onStop() {
        super.onStop()

        EventBus.getDefault().unregister(this)
    }
    
    override fun getData() {
        val parent = activity as SystemDetailsActivity
        parent.getData()
    }

    @Subscribe(threadMode = ThreadMode.MAIN)
    fun onStationsEvent(stations: SystemStations) {
        // Error case
        if (!stations.success) {
            endLoading(true)
            NotificationsUtils.displayGenericDownloadErrorSnackbar(activity)
            return
        }

        endLoading(stations.stations.isEmpty())
        recyclerViewAdapter.submitList(stations.stations)
    }
}
